### PR TITLE
Taskdata dir

### PR DESCRIPTION
**Pull request by mathstuf**

### CODE REVIEW

**Comment by ryneeverett** [comment]
What is the use case where the environment variable is inadequate?

File: bugwarrior/config.py [hunk 1]
```diff
@@ -185,7 +185,12 @@ def get_taskrc_path(conf, main_section):
 
 
 def get_data_path():
-    return os.path.expanduser(os.getenv('TASKDATA', '~/.task'))
+    path = os.getenv('TASKDATA', '~/.task')
+    if conf.has_option(main_section, 'taskdata'):
```
Comment (ryneeverett) [inline bugwarrior/config.py]: `conf` -> `config`

File: bugwarrior/config.py [hunk 2]
```diff
@@ -185,7 +185,12 @@ def get_taskrc_path(conf, main_section):
...
+        path = conf.get(main_section, 'taskdata')
```
Comment (ryneeverett) [inline bugwarrior/config.py]: s/conf/config/g

Comment (mathstuf) [inline bugwarrior/config.py]: It is `conf` up in `get_taskrc_path`. Is that wrong?

**Comment by mathstuf** [comment]
I can already set the `taskrc` in a per-flavor stanza; it'd be nice to handle the directory as well. Though I suppose asking `task` what it thinks the directory is would be the best route since that would just use the `taskrc`'s setting (and do the `TASKDATA` fallback that way).

**Comment by mathstuf** [comment]
Yeah, `task _show | grep '^data.location='` will give the location that task uses for its lists including `TASKDATA` in the computation.

File: bugwarrior/config.py (hunk 2 above)
Comment (ryneeverett) [inline bugwarrior/config.py]: That's a function parameter which doesn't exist in this function. Travis tests failed due to `NameError: global name 'conf' is not defined`.

Comment (mathstuf) [inline bugwarrior/config.py]: Oh, missed that it's passed in as an argument above. Probably mis-copied :/ . Anyways, not necessary for `task _show`.

Comment (ryneeverett) [inline bugwarrior/config.py]: I guess the config needs to be passed into this function for this to work.

**Comment by mathstuf** [comment]
Well, this is as far as I got tonight. Currently, the bitbucket test hangs for whatever reason, but I don't have pudb handy here and with all the fighting Python I did already today, I don't think being in one would be productive this late at night anyways.

There's probably a better design, but I've been doing Rust for the past week and Python feels…squishy now and I can't seem to get things quite right. I'll take another look tomorrow when I'll hopefully be able to see a better way of doing this.

**Comment by ryneeverett** [comment]
I took my own crack at this in #364.

**Comment by mathstuf** [comment]
Thanks; closing in favor of it. Much cleaner :) .

**Pull request closed without merging**